main.go

```diff
@@ -44,14 +47,52 @@ func mainWithError() (err error) {
 	// Define server factory to create the custom server once all command line
 	// flags are parsed and all microservice configuration is processed.
 	newServerFactory := func(v *viper.Viper) microserver.Server {
+		var restConfig *rest.Config
+		{
+			c := k8srestconfig.Config{
+				Logger: newLogger,
+
+				Address:    v.GetString(f.Service.Kubernetes.Address),
+				InCluster:  v.GetBool(f.Service.Kubernetes.InCluster),
+				KubeConfig: v.GetString(f.Service.Kubernetes.KubeConfig),
+				TLS: k8srestconfig.ConfigTLS{
+					CAFile:  v.GetString(f.Service.Kubernetes.TLS.CAFile),
+					CrtFile: v.GetString(f.Service.Kubernetes.TLS.CrtFile),
+					KeyFile: v.GetString(f.Service.Kubernetes.TLS.KeyFile),
+				},
+			}
+
+			restConfig, err = k8srestconfig.New(c)
+			if err != nil {
+				panic(err)
+			}
+		}
+
+		var k8sClient k8sclient.Interface
```

Comment:
Since both `server` and `service` need k8sclient object, creating a k8sclient object from `main.go` makes sense.